### PR TITLE
HDDS-8487. Recon Heatmap UI Changes and Size Normalization

### PR DESCRIPTION
**Pull request by smitajoshi12**
## What changes were proposed in this pull request?
HeatMap Changes
1) Size Normalization.
2) Color Change
3) Disabled Currentlt setting by flag.
4) Changed pnpm version to latest one

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-8487

## How was this patch tested?
Manually
image

image

image

image

### CODE REVIEW

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx]: @devmadhuu 
Completed Changes for  isHeatMapAvailable

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json]: Yes It will execute only isHeatMapAvailable flag set to true.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json
Comment (devmadhuu) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json:27]: what this is used for ? "heatmapTestClick"

**Review by devmadhuu — COMMENTED** [review]
@smitajoshi12 Looks most of the changes fine, just a minor comment. @smengcl - pls review

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx
Comment (devmadhuu) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx:38]: lets keep this as default "true" and show menu and UI, For empty response, don't render empty treemap, instead show "No Data Available" or some standard message if we are following any convention. Will document this in Ozone doc.

Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx]: @devmadhuu  Completed Changes.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json
Comment (dombizita) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json:27]: is this necessary? I can't see it used anywhere else

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/package.json
Comment (dombizita) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/package.json:41]: is this removed because of the previous **build (8)** issue? did you check the history why it was there? maybe @adoroszlai do you know anything about this?

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx
Comment (dombizita) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx:31]:
```suggestion
  isHeatmapAvailable: boolean;
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx
Comment (dombizita) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx:38]: I can't see that you did the changes suggested by @devmadhuu

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx
Comment (dombizita) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx:96]: shouldn't it be Heatmap? previously I saw it with this spelling.
cc @devmadhuu 
```suggestion
                <span>Heatmap</span>
```

**Review by dombizita — COMMENTED** [review]
thanks for the patch @smitajoshi12, I added a few comments, please take a look at them. also I'd suggest that someone with more frontend related knowledge should review this before merge.

**Comment by smitajoshi12** [comment]
@ivandika3 

Can you review my UI heatmap PR?

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/package.json
Comment (adoroszlai) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/package.json:41]: No, sorry, I don't know anything about this.

**Review by sumitagrawl — APPROVED** [review]
@smitajoshi12 LGTM +1, please check few comments from others if they are fine.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/package.json
Comment (sumitagrawl) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/package.json:41]: with removal, build was success as verified by smita

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMap.tsx
Comment (ivandika3) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMap.tsx:286]: Shall we move the `style={{ height: 800}}` to the `heatmapinformation` class in the `heatmap.less`?

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMap.tsx
Comment (ivandika3) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMap.tsx:236]: Nit: `menuCalendar`

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMap.tsx
Comment (ivandika3) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMap.tsx:327]: Let's move the CSS styles to `heatmap.less` if possible

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMapConfiguration.tsx
Comment (ivandika3) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMapConfiguration.tsx:74]: image

The bottom of the title and subtitles are truncated since the borders are also white. It also doesn't demarcate well with the white background. Maybe we can modify some of the colorings and distance between the border and the text?
I think the example in the website https://www.ag-grid.com/react-charts/gallery/market-index-treemap/ looks fine.

**Review by ivandika3 — CHANGES_REQUESTED** [review]
Hi @smitajoshi12, thank you for working on this. I have some suggestions.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMapConfiguration.tsx
Comment (devabhishekpal) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMapConfiguration.tsx:74]: Hello @ivandika3 , there is an alternative UI which I am working on, and will be pushed once this gets merged.

Comment (ivandika3) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMapConfiguration.tsx:74]: Noted, thanks.

Comment (ivandika3) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMapConfiguration.tsx:74]: @devabhishekpal My suggestion is to make the colorings adhere to the Recon's color scheme (e.g. the side bars color, Ozone logo color), so that the difference in colors are not too jarring.

**Comment by smitajoshi12** [comment]
@sumitagrawl  @dombizita @devmadhuu 
Completed all changes suggested by Ivan you can merge PR.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/constants/breadcrumbs.constants.tsx
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/constants/breadcrumbs.constants.tsx:30]: IMO should be lower case 'm':

```suggestion
  '/Heatmap': 'Heatmap'
```

but if it is too much effort to change it we file another jira

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/package.json
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/package.json:170]: Restore the extra line at the end

```suggestion
}

```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/routes.tsx
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/routes.tsx:26]: Consistency

```suggestion
import {DiskUsage} from './views/diskUsage/diskUsage';
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/routes.tsx
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/routes.tsx:27]: if you want to do the rename in this PR
```suggestion
import {Heatmap} from './views/heatmap/heatmap';
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/routes.tsx
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/routes.tsx:64]: if you want to do the rename in this PR
```suggestion
    path: '/Heatmap',
    component: Heatmap
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx:24]:
```suggestion
import {RouteComponentProps} from 'react-router';
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx:126]: if you want to do the rename in this PR
```suggestion
              <Menu.Item key='/Heatmap'>
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/navBar/navBar.tsx:129]: if you want to do the rename in this PR
```suggestion
                <Link to='/Heatmap' />
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMap.tsx
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMap.tsx:1]: if you want to do the rename in this PR, file path should be:

```
views/heatmap/heatmap.tsx
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMap.tsx
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMap.tsx:57]: same here

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatmap.less
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatmap.less:60]: add new line at the end
```suggestion
}

```

**Review by smengcl — APPROVED** [review]
lgtm apart a few nits

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMapConfiguration.tsx
Comment (devabhishekpal) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/heatMap/heatMapConfiguration.tsx:75]: Will keep this in mind

**Comment by smitajoshi12** [comment]
Completed all changes suggested by Siyao.

**Pull request merged**
